mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/store/ArtifactID.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.server.master.store;
+
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+
+/**
+ * Represents the job artifact that mantis can run.
+ */
+@RequiredArgsConstructor(staticName="of")
+@Value
+public class ArtifactID {
```

Comment:
Shouldn't this be part of the client? I would assume you would want to expose APIs around which artifacts are available in mantis, used by which job clusters, and so on.